src/api/apiMethod.js

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright © 2017 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+
+import { GET } from 'constants';
+import { solveURLParams, toQueryString } from './utils';
+
+// Bind to resource class
+const apiMethod = (params = {}) => {
+	const {
+		method = GET,
+		path = '',
+		urlParams = [],
+		validator = null,
+		defaultData = {},
+		retry = false,
+	} = params;
+
+	return function apiHandler(...args) {
+		if (urlParams.length > 0 && args.length < urlParams.length) {
+			return Promise.reject(
+				new Error(
+					'Arguments must include Params defined. Required: ',
+					urlParams.toString(),
+				),
+			);
+		}
+
+		const data = Object.assign(
+			{},
+			defaultData,
+			args.length > urlParams.length &&
+			typeof args[urlParams.length] === 'object'
```

Comment:
What if there are additional args beyond `urlParams.length`?

Reply:
Not used!
Always, args[urlPrams.length] is the body or query object.